### ISSUE TITLE
Create operation doesn't work for remoting

### ISSUE THREAD

**Issue by Wayenyin** [Microsoft/omi-script-provider#66]
**Build**

- omi path:[\\redmond\wsscfs\OSTCData\Builds\omi\develop\1.4.0-37 ](\\redmond\wsscfs\OSTCData\Builds\omi\develop\1.4.0-37 )
- Python script:[\\redmond\wsscfs\OSTCData\Builds\omiscriptprovider\develop\1.1.1-35 ](\\redmond\wsscfs\OSTCData\Builds\omiscriptprovider\develop\1.1.1-35 )

**Reproduce steps:**
1.	Install omi package successfully.
2.	Install Python provider package successfully.
3.	Add port 5985 to /etc/opt/omi/conf/omiserver.conf file on Linux, then run command "_/opt/omi/bin/service_control restart_"; 
4.	Upload [Host provider](https://github.com/Microsoft/omi-script-provider/blob/master/samples/hosts/_mi_main.py) to /opt/omi/lib folder on Linux, then register Host provider
Run command:” _cd  /opt/omi/lib;  /opt/omi/bin/omireg --Python hosts_”
```
   /opt/omi/lib
   root@omi64-cent7-01  # /opt/omi/bin/omireg --Python hosts
   Created /etc/opt/omi/conf/omiregister/root-cimv2/hosts.reg
   /opt/omi/bin/omiserver: refreshed server
```
5.	Execute created command and return error message:
```
   root@omi64-cent7-01  # /opt/omi/bin/omicli ci  -u root -p *** --auth Basic --hostname omi64-cent7-01 root/cimv2 { Hosts FQDN 'omi64-cent7-01.scx.com' IPAddress '10.226.158.99' } --port 5985 --encryption none
   /opt/omi/bin/omicli: result: MI_RESULT_FAILED
   /opt/omi/bin/omicli: result: ERROR_INTERNAL_ERROR: unexpected internal state
```
6.	Verify result, but the new instance has been created successfully:
```
   root@omi64-cent7-01  # /opt/omi/bin/omicli ei  -u root -p *** --auth Basic --hostname omi64-cent7-01 root/cimv2  Hosts  --port 5985 --encryption none
   instance of Hosts
   {
        FQDN=localhost localhost.localdomain localhost4 localhost4.localdomain4
        IPAddress=127.0.0.1
   }
   instance of Hosts
   {
        FQDN=localhost localhost.localdomain localhost6 localhost6.localdomain6
        IPAddress=::1
   }
   instance of Hosts
   {
        FQDN=omi64-cent7-01.scx.com
        IPAddress=10.226.158.99
   }
```
**Expect result:** Step 5 should get pass and no error.
**Actual result:** Step 5 failed with error.

**Note:** If we just run create for Local, it will work and not return error, the details show as below:

-  Check Host provider with 'ei' command
```
   /opt/omi/lib
   root@omi64-cent7-01  # /opt/omi/bin/omicli ei  root/cimv2  Hosts
   instance of Hosts
   {
        [Key] FQDN=localhost localhost.localdomain localhost4 localhost4.localdomain4
        IPAddress=127.0.0.1
   }
   instance of Hosts
   {
        [Key] FQDN=localhost localhost.localdomain localhost6 localhost6.localdomain6
        IPAddress=::1
   }
```
- Run 'ci' command:
```
   /opt/omi/lib
   root@omi64-cent7-01  # /opt/omi/bin/omicli ci  root/cimv2 { Hosts FQDN 'omi64-cent7-01.scx.com' IPAddress '10.226.158.99' }
```

- Check Host provider again:
```
   /opt/omi/lib
   root@omi64-cent7-01  # /opt/omi/bin/omicli ei  root/cimv2  Hosts
   instance of Hosts
   {
        [Key] FQDN=localhost localhost.localdomain localhost4 localhost4.localdomain4
        IPAddress=127.0.0.1
   }
   instance of Hosts
   {
        [Key] FQDN=localhost localhost.localdomain localhost6 localhost6.localdomain6
        IPAddress=::1
   }
   instance of Hosts
   {
        [Key] FQDN=omi64-cent7-01.scx.com
        IPAddress=10.226.158.99
   }
```

**Comment by EMumau** [comment]
These errors are part of OMI server functionality and are outside of the scope of functionality of the script provider.  All authorization is part of the OMI server.  Providers have nothing to do with authentication.

**Comment by JumpingYang001** [comment]
OK, but other operations works fine for remoting, if we don't support remoting for the script provider, we need to document it, and we might not test remoting in this release, thanks.

**Comment by paulcallen** [comment]
we do support remoting, but we need to understand why this is failing. This provider is treated as just another provider. 
Does this fail if you use the binary protocol locally (do local operation, don't specify host, no port, no authentication)?

**Comment by JumpingYang001** [comment]
@paulcallen , local works fine:
```
   root@omi64-cent7-01  # /opt/omi/bin/omicli ci  root/cimv2 { Hosts FQDN 'omi64-cent7-01.scx.com' IPAddress '10.226.158.99' }
```

**Comment by JumpingYang001** [comment]
Remoting command:
```
/opt/omi/bin/omicli ci  -u root -p ****** --auth Basic --hostname localhost root/cimv2 { Hosts FQDN 'omi64-cent7-01.scx.com' IPAddress '10.226.158.99' } --port 5985 --encryption none
/opt/omi/bin/omicli: result: MI_RESULT_FAILED
/opt/omi/bin/omicli: result: ERROR_INTERNAL_ERROR: unexpected internal state
```

miclient.log:
```
2017/11/14 20:25:03 [10359,10359] ERROR: null(0): _CreateSocketAndConnect - Sock_Connect failed. result: 1 (MI_RESULT_FAILED)
2017/11/14 20:25:03 [10359,10359] WARNING: null(0): _CreateConnectorSocket - _CreateSocketAndConnect of primary address failed. result: 1 (MI_RESULT_FAILED)
2017/11/14 20:25:03 [10359,10359] ERROR: null(0): EventId=20106 Priority=ERROR Trying to thunk generic handle that has been shutdown already: 0x7fff1bfd75e0
2017/11/14 20:25:03 [10359,10360] ERROR: null(0): _RequestCallback - RequestCallbackRead failed
```

omiclient-recv.trc:
```
HTTP/1.1 500 Internal Server Error
Content-Length: 1394
Connection: Keep-Alive
Content-Type: application/soap+xml;charset=UTF-8

<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://www.w3.org/2003/05/soap-envelope" xmlns:wsa="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:wsen="http://schemas.xmlsoap.org/ws/2004/09/enumeration" xmlns:e="http://schemas.xmlsoap.org/ws/2004/08/eventing" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:wsmb="http://schemas.dmtf.org/wbem/wsman/1/cimbinding.xsd" xmlns:wsman="http://schemas.dmtf.org/wbem/wsman/1/wsman.xsd" xmlns:wxf="http://schemas.xmlsoap.org/ws/2004/09/transfer" xmlns:cim="http://schemas.dmtf.org/wbem/wscim/1/common" xmlns:msftwinrm="http://schemas.microsoft.com/wbem/wsman/1/wsman.xsd" xmlns:wsmid="http://schemas.dmtf.org/wbem/wsman/identity/1/wsmanidentity.xsd"><SOAP-ENV:Header><wsa:To>http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</wsa:To><wsa:Action>http://schemas.dmtf.org/wbem/wsman/1/wsman/fault</wsa:Action><wsa:MessageID>uuid:E205B88B-5DFD-0005-0000-000000020000</wsa:MessageID><wsa:RelatesTo>uuid:E32FAE59-5DFD-0005-0000-000000010000</wsa:RelatesTo></SOAP-ENV:Header><SOAP-ENV:Body><SOAP-ENV:Fault><SOAP-ENV:Code><SOAP-ENV:Value>SOAP-ENV:Receiver</SOAP-ENV:Value><SOAP-ENV:Subcode><SOAP-ENV:Value>wsman:InternalError</SOAP-ENV:Value></SOAP-ENV:Subcode></SOAP-ENV:Code><SOAP-ENV:Reason><SOAP-ENV:Text xml:lang="en-US">unexpected internal state</SOAP-ENV:Text></SOAP-ENV:Reason></SOAP-ENV:Fault></SOAP-ENV:Body></SOAP-ENV:Envelope>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://www.w3.org/2003/05/soap-envelope" xmlns:wsa="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:wsen="http://schemas.xmlsoap.org/ws/2004/09/enumeration" xmlns:e="http://schemas.xmlsoap.org/ws/2004/08/eventing" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:wsmb="http://schemas.dmtf.org/wbem/wsman/1/cimbinding.xsd" xmlns:wsman="http://schemas.dmtf.org/wbem/wsman/1/wsman.xsd" xmlns:wxf="http://schemas.xmlsoap.org/ws/2004/09/transfer" xmlns:cim="http://schemas.dmtf.org/wbem/wscim/1/common" xmlns:msftwinrm="http://schemas.microsoft.com/wbem/wsman/1/wsman.xsd" xmlns:wsmid="http://schemas.dmtf.org/wbem/wsman/identity/1/wsmanidentity.xsd"><SOAP-ENV:Header><wsa:To>http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</wsa:To><wsa:Action>http://schemas.dmtf.org/wbem/wsman/1/wsman/fault</wsa:Action><wsa:MessageID>uuid:E205B88B-5DFD-0005-0000-000000020000</wsa:MessageID><wsa:RelatesTo>uuid:E32FAE59-5DFD-0005-0000-000000010000</wsa:RelatesTo></SOAP-ENV:Header><SOAP-ENV:Body><SOAP-ENV:Fault><SOAP-ENV:Code><SOAP-ENV:Value>SOAP-ENV:Receiver</SOAP-ENV:Value><SOAP-ENV:Subcode><SOAP-ENV:Value>wsman:InternalError</SOAP-ENV:Value></SOAP-ENV:Subcode></SOAP-ENV:Code><SOAP-ENV:Reason><SOAP-ENV:Text xml:lang="en-US">unexpected internal state</SOAP-ENV:Text></SOAP-ENV:Reason></SOAP-ENV:Fault></SOAP-ENV:Body></SOAP-ENV:Envelope>
```
miclient-send.trc
```
POST /wsman/ HTTP/1.1
Connection: Keep-Alive
Content-Length: 1419
Content-Type: application/soap+xml;charset=UTF-8
Authorization: Basic cm9vdDowNWVPc0deMTc=
Host: localhost:5985


<s:Envelope xmlns:s="http://www.w3.org/2003/05/soap-envelope" xmlns:a="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:n="http://schemas.xmlsoap.org/ws/2004/09/enumeration" xmlns:w="http://schemas.dmtf.org/wbem/wsman/1/wsman.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema" xmlns:h="http://schemas.microsoft.com/wbem/wsman/1/windows/shell" xmlns:p="http://schemas.microsoft.com/wbem/wsman/1/wsman.xsd" ><s:Header><a:To>HTTP://localhost:5985/wsman/</a:To><w:ResourceURI s:mustUnderstand="true">http://schemas.dmtf.org/wbem/wscim/1/cim-schema/2/Hosts</w:ResourceURI><a:ReplyTo><a:Address s:mustUnderstand="true">http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</a:Address></a:ReplyTo><a:Action>http://schemas.xmlsoap.org/ws/2004/09/transfer/Create</a:Action><w:MaxEnvelopeSize s:mustUnderstand="true">102400</w:MaxEnvelopeSize><a:MessageID>uuid:E32FAE59-5DFD-0005-0000-000000010000</a:MessageID><w:OperationTimeout>PT1M30S</w:OperationTimeout><w:Locale xml:lang="en-us" s:mustUnderstand="false"/><p:DataLocale xml:lang="en-us" s:mustUnderstand="false"/><w:OptionSet s:mustUnderstand="true"></w:OptionSet><w:SelectorSet><w:Selector Name="__cimnamespace">root/cimv2</w:Selector></w:SelectorSet></s:Header><s:Body><p:Hosts xmlns:p="http://schemas.dmtf.org/wbem/wscim/1/cim-schema/2/Hosts"><p:FQDN>omi64-cent7-02.scx.com</p:FQDN><p:IPAddress>10.226.158.99</p:IPAddress></p:Hosts></s:Body></s:Envelope>
```
omiserver-recv.trc
```
ⴊⴭⴭⴭⴭⴭ‭晁整⁲敄牣灹楴湯ⴠⴭⴭⴭⴭⴭⴭⴭ
㰊㩳湅敶潬数砠汭獮猺∽瑨灴⼺眯睷眮⸳牯⽧〲㌰〯⼵潳灡攭癮汥灯≥砠汭獮愺∽瑨灴⼺猯档浥獡砮汭潳灡漮杲眯⽳〲㐰〯⼸摡牤獥楳杮•浸湬㩳㵮栢瑴㩰⼯捳敨慭⹳浸獬慯⹰牯⽧獷㈯〰⼴㤰支畮敭慲楴湯•浸湬㩳㵷栢瑴㩰⼯捳敨慭⹳浤晴漮杲眯敢⽭獷慭⽮⼱獷慭⹮獸≤砠汭獮砺楳∽瑨灴⼺眯睷眮⸳牯⽧〲㄰堯䱍捓敨慭•浸湬㩳㵨栢瑴㩰⼯捳敨慭⹳業牣獯景⹴潣⽭扷浥眯浳湡ㄯ眯湩潤獷猯敨汬•浸湬㩳㵰栢瑴㩰⼯捳敨慭⹳業牣獯景⹴潣⽭扷浥眯浳湡ㄯ眯浳湡砮摳•㰾㩳效摡牥㰾㩡潔䠾呔㩐⼯潬慣桬獯㩴㤵㔸眯浳湡㰯愯吺㹯眼刺獥畯捲啥䥒猠洺獵啴摮牥瑳湡㵤琢畲≥栾瑴㩰⼯捳敨慭⹳浤晴漮杲眯敢⽭獷楣⽭⼱楣⵭捳敨慭㈯䠯獯獴⼼㩷敒潳牵散剕㹉愼刺灥祬潔㰾㩡摁牤獥⁳㩳畭瑳湕敤獲慴摮∽牴敵㸢瑨灴⼺猯档浥獡砮汭潳灡漮杲眯⽳〲㐰〯⼸摡牤獥楳杮爯汯⽥湡湯浹畯㱳愯䄺摤敲獳㰾愯刺灥祬潔㰾㩡捁楴湯栾瑴㩰⼯捳敨慭⹳浸獬慯⹰牯⽧獷㈯〰⼴㤰琯慲獮敦⽲牃慥整⼼㩡捁楴湯㰾㩷慍䕸癮汥灯卥穩⁥㩳畭瑳湕敤獲慴摮∽牴敵㸢〱㐲〰⼼㩷慍䕸癮汥灯卥穩㹥愼䴺獥慳敧䑉甾極㩤㍅䘲䕁㤵㔭䙄ⵄ〰㔰〭〰ⴰ〰〰〰㄰〰〰⼼㩡敍獳条䥥㹄眼伺数慲楴湯楔敭畯㹴呐䴱〳㱓眯伺数慲楴湯楔敭畯㹴眼䰺捯污⁥浸㩬慬杮∽湥甭≳猠洺獵啴摮牥瑳湡㵤昢污敳⼢㰾㩰慄慴潌慣敬砠汭氺湡㵧攢⵮獵•㩳畭瑳湕敤獲慴摮∽慦獬≥㸯眼伺瑰潩卮瑥猠洺獵啴摮牥瑳湡㵤琢畲≥㰾眯伺瑰潩卮瑥㰾㩷敓敬瑣牯敓㹴眼区汥捥潴⁲慎敭∽彟楣湭浡獥慰散㸢潲瑯振浩㉶⼼㩷敓敬瑣牯㰾眯区汥捥潴卲瑥㰾猯䠺慥敤㹲猼䈺摯㹹瀼䠺獯獴砠汭獮瀺∽瑨灴⼺猯档浥獡搮瑭⹦牯⽧扷浥眯捳浩ㄯ振浩猭档浥⽡⼲潈瑳≳㰾㩰兆乄漾業㐶挭湥㝴〭⸲捳⹸潣㱭瀯䘺䑑㹎瀼䤺䅐摤敲獳ㄾ⸰㈲⸶㔱⸸㤹⼼㩰偉摁牤獥㹳⼼㩰潈瑳㹳⼼㩳潂祤㰾猯䔺癮汥灯㹥ਊⴭⴭⴭⴭⴭⴭⴭ䔠摮䐠捥祲瑰ⴠⴭⴭⴭⴭⴭⴭⴭⴭਭ਀
```

**Comment by JumpingYang001** [comment]
WSMan with local works fine, too.
```
root@omi64ub16-dev1:/opt/omi/lib# /opt/omi/bin/omicli ci  -u root -p 05eOsG^17  root/cimv2 { Hosts FQDN '
omi64-cent7-03.scx.com' IPAddress '10.226.158.99' }
```

**Comment by JumpingYang001** [comment]
WSMan with local works fine, too.
```
root@omi64ub16-dev1:/opt/omi/lib# /opt/omi/bin/omicli ci  -u root -p ******  root/cimv2 { Hosts FQDN '
omi64-cent7-03.scx.com' IPAddress '10.226.158.99' }
```
I think above operation is using "operation locally using wsman protocol", right?

**Comment by JumpingYang001** [comment]
Binary with local works fine:
```
root@omi64ub16-dev1:/opt/omi/lib# /opt/omi/bin/omicli ci root/cimv2 { Hosts FQDN 'omi64-cent7-04.scx.com'
 IPAddress '10.226.158.99' }
```

**Comment by JumpingYang001** [comment]
I want to point out that ei works fine with remoting, but ci failed with remoting:
```
root@omi64ub16-dev1:/opt/omi/lib# /opt/omi/bin/omicli ei  -u root -p ****** --auth Basic --hostname localhost root/cimv2 Hosts --port 5985 --encryption none                                                   
instance of Hosts
{
    FQDN=localhost
    IPAddress=127.0.0.1
}
instance of Hosts
{
    FQDN=omi64ub16-dev1
    IPAddress=10.226.210.177
}
instance of Hosts
{
    FQDN=localhost ip6-localhost ip6-loopback
    IPAddress=::1
}
instance of Hosts
{
    FQDN=ip6-allnodes
    IPAddress=ff02::1
}
instance of Hosts
{
    FQDN=ip6-allrouters
    IPAddress=ff02::2
}
instance of Hosts
{
    FQDN=omi64-cent7-01.scx.com
    IPAddress=10.226.158.99
}
instance of Hosts
{
    FQDN=omi64-cent7-02.scx.com
    IPAddress=10.226.158.99
}
instance of Hosts
{
    FQDN=omi64-cent7-03.scx.com
    IPAddress=10.226.158.99
}
instance of Hosts
{
    FQDN=omi64-cent7-04.scx.com
    IPAddress=10.226.158.99
}
instance of Hosts
{
    FQDN=omi64-cent7-05.scx.com
    IPAddress=10.226.158.99
}
instance of Hosts
{
    FQDN=omi64-cent7-06.scx.com
    IPAddress=10.226.158.99
}
root@omi64ub16-dev1:/opt/omi/lib# 
```

**Comment by JumpingYang001** [comment]
The provider we used is here: https://github.com/Microsoft/omi-script-provider/blob/master/samples/hosts/_mi_main.py

**Comment by JumpingYang001** [comment]
omiagent.root.root.log:
```
2017/11/14 21:10:06 [12834,12834] INFO: null(0): EventId=40032 Priority=INFO Selector_AddHandler: selector=0x5bb2b8, handler=0x155b710, name=BINARY_FROM_SOCKET
2017/11/14 21:10:06 [12834,12834] INFO: null(0): EventId=40032 Priority=INFO Selector_AddHandler: selector=0x5bb2b8, handler=0x5bb220, name=PROVMGR_TIMEOUT_MANAGER
2017/11/14 21:10:06 [12834,12834] INFO: null(0): EventId=40003 Priority=INFO agent started; fd 8
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45005 Priority=DEBUG MessageFromBatch:- msg tag is 15, index: 15
2017/11/14 21:10:06 [12834,12834] INFO: null(0): EventId=40011 Priority=INFO done with receiving msg(0x155d248:15:BinProtocolNotification:1)
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45367 Priority=DEBUG Handle:(0x155b710), ClientAuthState = 4, EngineAuthState = 4
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45057 Priority=DEBUG _ConnectionIn_Post: 0x155dac0 Post msg(0x155d248:15:BinProtocolNotification:1) for interaction [0x155db58]<-0x155b780
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45055 Priority=DEBUG _ConnectionIn_FindRequest, Agent 0x155dac0(0x155db20), Cannot find key: 1 (may be new request)
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45058 Priority=DEBUG _ConnectionIn_Post: New operation with operationId (0x1), msg (0x155d248:BinProtocolNotification)
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45151 Priority=DEBUG ProtocolSocket: Ack on interaction [0x155b780]<-0x155db58
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45005 Priority=DEBUG MessageFromBatch:- msg tag is 4109, index: 13
2017/11/14 21:10:06 [12834,12834] INFO: null(0): EventId=40011 Priority=INFO done with receiving msg(0x155dfe8:4109:CreateInstanceReq:2)
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45367 Priority=DEBUG Handle:(0x155b710), ClientAuthState = 4, EngineAuthState = 4
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45057 Priority=DEBUG _ConnectionIn_Post: 0x155dac0 Post msg(0x155dfe8:4109:CreateInstanceReq:2) for interaction [0x155db58]<-0x155b780
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45055 Priority=DEBUG _ConnectionIn_FindRequest, Agent 0x155dac0(0x155db20), Cannot find key: 2 (may be new request)
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45058 Priority=DEBUG _ConnectionIn_Post: New operation with operationId (0x2), msg (0x155dfe8:CreateInstanceReq)
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45151 Priority=DEBUG ProtocolSocket: Ack on interaction [0x155b780]<-0x155db58
2017/11/14 21:10:06 [12834,12834] INFO: null(0): EventId=40039 Priority=INFO New request received: command=(CreateInstanceReq), namespace=(root/cimv2), class=(Hosts)
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45173 Priority=DEBUG _GetProviderByClassName{Hosts}
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45072 Priority=DEBUG Context: New context 0x7ffe1bcf0120 with interaction: (nil)->[0x7ffe1bcf0180]
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45160 Priority=DEBUG post result from provider, 0
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45071 Priority=DEBUG Context: Destroy 0x7ffe1bcf0120 (strand: 0x7ffe1bcf0148, interaction [0x7ffe1bcf0180])
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45072 Priority=DEBUG Context: New context 0x7ffe1bcf0120 with interaction: (nil)->[0x7ffe1bcf0180]
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45160 Priority=DEBUG post result from provider, 0
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45071 Priority=DEBUG Context: Destroy 0x7ffe1bcf0120 (strand: 0x7ffe1bcf0148, interaction [0x7ffe1bcf0180])
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45072 Priority=DEBUG Context: New context 0x155fd80 with interaction: 0x155d2f8->[0x155fde0]
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45073 Priority=DEBUG OperationOut: Ack on interaction [0x155d2f8]<-0x155fde0
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45160 Priority=DEBUG post result from provider, 0
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45063 Priority=DEBUG Context_PostMessageLeft_IoThread: 0x155fd80
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45060 Priority=DEBUG _Context_Aux_PostLeft: 0x155fda8, thisAckPending: 0
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45148 Priority=DEBUG ProtocolSocket: Posting message for interaction [0x155b780]<-0x155db58
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45206 Priority=DEBUG Sending msg(0x15631a8:4:PostResultMsg:2) on own thread
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45154 Priority=DEBUG Socket: 0x155b710, All send
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45076 Priority=DEBUG ConnectionIn: Ack on interaction [0x155db58]<-0x155b780
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45070 Priority=DEBUG Context: Ack on context 0x155fda8, interaction 0x155d2f8->[0x155fde0], (Transport Closed: ByThis: 0, ByOther: 0), tryingToPostLeft: 3
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45062 Priority=DEBUG _Context_Aux_PostLeft_Notify(IoThread): 0x155fda8
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45075 Priority=DEBUG OperationOut: Close on interaction [0x155d2f8]<-0x155fde0
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45068 Priority=DEBUG _Context_Close: Close on context 0x155fda8, interaction 0x155d2f8->[0x155fde0]
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45069 Priority=DEBUG Context: 0x155fda8, finish called
2017/11/14 21:10:06 [12834,12834] DEBUG: null(0): EventId=45071 Priority=DEBUG Context: Destroy 0x155fd80 (strand: 0x155fda8, interaction [0x155fde0])
```

**Comment by JumpingYang001** [comment]
from miclient.log, it failed at : https://github.com/Microsoft/omi/blob/1770fbeb6eb52e32781d5efb0918911525134486/Unix/http/httpclient.c#L1702

**Comment by JumpingYang001** [comment]
`/opt/omi/bin/omicli: result: ERROR_INTERNAL_ERROR: unexpected internal state`
stack 1: 
https://github.com/Microsoft/omi/blob/661d311e5156955c1804c901d403ba6f514a4cbf/Unix/wsman/wsman.c#L3283
stack 2: 
https://github.com/Microsoft/omi/blob/661d311e5156955c1804c901d403ba6f514a4cbf/Unix/wsman/wsman.c#L3751

`gdb attach pidofomiengine`

**Comment by JumpingYang001** [comment]
@paulcallen , I am not familiar with selfCD->wsheader.rqtAction, its value in above "stack 1" link limited to PUT and DELETE, maybe you know why it is not the PUT or DELETE here? :)

**Comment by JumpingYang001** [comment]
BTW, the ci operation with remoting and omicli get MI_RESULT_FAILED, but we can ei operation with remoting and we can find the ci instance is created and it shows in the ei result.

**Comment by JumpingYang001** [comment]
I was trying to get selfCD->wsheader.rqtAction value with omi-1.4.1-0.ssl_100.ulinux.x64.withsymbols.deb, but gdb shows it is optimized value which I cannot get:
```
(gdb) break wsman.c:3267
Breakpoint 1 at 0x40c1b7: file wsman.c, line 3267.
(gdb) c
Continuing.

Breakpoint 1, _CD_RightCloseHandler (self=<optimized out>) at wsman.c:3267
3267    wsman.c: No such file or directory.
(gdb) p self->wsheader.rqtAction                                                                        
value has been optimized out

```

**Comment by paulcallen** [comment]
@JumpingYang001 There are a few things going on in the description here that needs to be clarified:
- omicli needs a --hostname in order to use WSMAN protocol. If you use this locally you will get the same error. Therefore the error is coming from inside the wsman layer. If I had known this was the case it would have made diagnostics a little easier.
- This error 'unexpected internal state' actually only comes from 2 places
   - _ProcessEmptyBodyResponse
   - _ProcessInstanceResponse
It sends this error if an operation that should get an instance and does not get one, or an operation that should not get an instance and does get one. 
- Create Instance *MUST* have an instance posted back for the WSMAN protocol, and *SHOULD* be sent back over the binary protocol, but the binary protocol does not complain if it is not there

Therefore we have possibly 2 places that need to be fixed:
- The Host provider *MUST* return an instance back after it finishes. It can even be the one that was sent in to the create. Sometimes there are read-only properties, or other properties that may need to be added before returning, but there needs to be an instance
- The script provider needs to support the posting back of an instance for Create Instance, don't know if it is supported or not right now though.

Conclusion, this is not an OMI issue. It may be a provider and/or script provider issue.

**Comment by JumpingYang001** [comment]
@paulcallen  ,in fact, the script provider doesn't support post instance, here is the issue #65 which was closed.
And the #65 was marked as won't fix/by design by @EMumau, so can you have a talk with Eric for this issue to address it and make a good solution? thank you!

**Comment by JumpingYang001** [comment]
If we want to fix all 15 platforms compiling issue, that might need much more time.
Our current highest priority for issues is these 2 issues:
 - the create operation with remoting issue 
 - the python process memory leak issue.

**Comment by EMumau** [comment]
I'll change the omi script provider to support post instance today.  I'll report back when that is complete.

**Comment by EMumau** [comment]
This should be resolved in PR #77

**Comment by Wayenyin** [comment]
This issue has been fixed in Python provider build 1.1.1-47.
```
root@omi64-cent7-01  # /opt/omi/bin/omicli ci -u root -p 05e*** --auth Basic --hostname omi64-cent7-01 root/cimv2 { TestClass_AllDMTFArrays v_Key 5 } --port 5985 --encryption none
instance of TestClass_AllDMTFArrays
{
    [Key] v_Key=5
}

```
